plugins/http_plugin/http_plugin.cpp

```diff
@@ -448,7 +449,7 @@ namespace eosio {
           */
          template<typename T>
          static auto make_in_flight(T&& object, http_plugin_impl& impl) {
-            return in_flight<T>(std::forward<T>(object), impl);
+            return std::make_shared<in_flight<T>>(in_flight<T>(std::forward<T>(object), impl));
```

Comment:
Does this have to be a `shared_ptr`? Can't it just be moved into `wrapped_then`  then when `wrapped_then` is destructed it will as well.

Reply:
> Does this have to be a `shared_ptr`? Can't it just be moved into `wrapped_then` then when `wrapped_then` is destructed it will as well.

No. Since in_flight disables its copy constructor, tracked_b gets deleted if it's not wrapped by a shared pointer.

Reply:
Move it, not copy it.

Reply:
> Move it, not copy it.

It failed to move.

Reply:
there are a few reasons why we moved this to a `std::shared_ptr` the first is that it makes it copyable which is necessary for use with `std::function`.  The second is that we have multiple asynchronous paths so there was no single lifecycle for `tracked_b`.  We are leaning on the `std::shared_ptr` 's reference counting to give us the lifetime guarantees we want.